packages/rspack/src/builtin-plugin/lazy-compilation/middleware.ts

```diff
@@ -37,19 +37,104 @@ const getFullServerUrl = ({ serverUrl, prefix }: LazyCompilationOptions) => {
 };
 
 export const lazyCompilationMiddleware = (
-	compiler: Compiler,
-	userOptions: LazyCompilationOptions | boolean = {}
+	compiler: Compiler | MultiCompiler,
+	rawOptions?: LazyCompilationOptions | boolean
 ): Middleware => {
+	let userOptions = rawOptions;
 	if (userOptions === false) {
 		return noop;
 	}
 
-	const options = userOptions === true ? {} : userOptions;
+	if (userOptions === true) {
+		userOptions = {};
+	}
+
+	if (compiler instanceof MultiCompiler) {
+		const middlewareByCompiler: Map<string, Middleware> = new Map();
+
+		let i = 0;
+		for (const c of compiler.compilers) {
+			if (!c.options.experiments.lazyCompilation) {
+				continue;
+			}
+
+			const options = {
+				...c.options.experiments.lazyCompilation,
+				...userOptions
+			};
+
+			const prefix = options.prefix || LAZY_COMPILATION_PREFIX;
+			options.prefix = `${prefix}__${i++}`;
+			const activeModules = new Map<string, boolean>();
+			const filesByKey = new Map<string, string>();
+
+			middlewareByCompiler.set(
+				options.prefix,
+				lazyCompilationMiddlewareInternal(
+					compiler,
+					activeModules,
+					filesByKey,
+					options.prefix
+				)
+			);
+
+			const compilerConfig = c.options.experiments.lazyCompilation
+				? c.options.experiments.lazyCompilation
+				: {};
+
+			const testConfig = compilerConfig.test ?? options.test;
+
+			const plugin = new BuiltinLazyCompilationPlugin(
+				({ module, path }) => {
```

Comment:
Can we extract a function to reduce duplicated code:

- https://github.com/web-infra-dev/rspack/blob/83046475af1e94b20dbfb79458ef104f150c76f0/packages/rspack/src/builtin-plugin/lazy-compilation/middleware.ts#L88-L103
- https://github.com/web-infra-dev/rspack/blob/83046475af1e94b20dbfb79458ef104f150c76f0/packages/rspack/src/builtin-plugin/lazy-compilation/middleware.ts#L138-L153